web/src/main/java/org/fao/geonet/monitor/link/LinkMonitorService.java

```diff
@@ -0,0 +1,142 @@
+package org.fao.geonet.monitor.link;
+
+import jeeves.interfaces.Logger;
+import jeeves.server.resources.ResourceManager;
+import jeeves.utils.Log;
+import org.apache.lucene.document.DocumentStoredFieldVisitor;
+import org.fao.geonet.GeonetContext;
+import org.fao.geonet.constants.Geonet;
+import org.fao.geonet.kernel.search.IndexAndTaxonomy;
+import org.fao.geonet.kernel.search.index.GeonetworkMultiReader;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class LinkMonitorService extends LinkMonitorInterface {
+
+    public enum Status {
+        FAILED,
+        WORKING,
+        UNKNOWN
+    }
+
+    public static int percentWorking = 90;
```

Comment:
Yeah, something I wrote in the email this morning, so... Yes, will be moved to config.